### PR TITLE
Add node maintenance activity

### PR DESCRIPTION
**Pull request by rawagner**

### CODE REVIEW

File: frontend/packages/metal3-plugin/src/plugin.ts [hunk 1]
```diff
@@ -138,6 +141,23 @@ const plugin: Plugin<ConsumedExtensions> = [
...
+        kind: referenceForModel(NodeMaintenanceModel),
+        prop: 'maintenances',
+      },
+      isActivity: (resource) => _.get(resource.status, 'phase') !== 'Succeeded',
+      getTimestamp: (resource) => new Date(resource.metadata.creationTimestamp),
+      loader: () =>
+        import(
+          './components/maintenance/MaintenanceDashboardActivity' /* webpackChunkName: "node-maintenance" */
+        ).then((m) => m.default),
+      required: FLAGS.BAREMETAL,
```
Comment (jtomasek) [inline frontend/packages/metal3-plugin/src/plugin.ts]:
```suggestion
      required: [FLAGS.BAREMETAL, Metal3_FLAG],
```

File: frontend/packages/metal3-plugin/src/plugin.ts [hunk 2]
```diff
@@ -138,6 +141,23 @@ const plugin: Plugin<ConsumedExtensions> = [
...
   },
+  {
+    type: 'Dashboards/Overview/Activity/Resource',
+    properties: {
+      k8sResource: {
+        isList: true,
+        kind: referenceForModel(NodeMaintenanceModel),
+        prop: 'maintenances',
+      },
+      isActivity: (resource) => _.get(resource.status, 'phase') !== 'Succeeded',
```
Comment (jtomasek) [inline frontend/packages/metal3-plugin/src/plugin.ts]: IIUC the maintenance phase can be `Running|Success|Error` so I'd go for 
```suggestion
      isActivity: (resource) => _.get(resource.status, 'phase') === 'Running',
```
just to be sure.

**Comment by rawagner** [comment]
@jtomasek fixed both. I think we will want to show maintenance progress in Status card for BMH dashboard. I will create a separate PR

**Pull request merged**